src/ScriptBuilder/Saga/Writers/OracleSagaScriptWriter.cs

```diff
@@ -0,0 +1,214 @@
+﻿using System;
+using System.IO;
+using System.Security.Cryptography;
+using System.Text;
+using NServiceBus.Persistence.Sql.ScriptBuilder;
+
+class OracleSagaScriptWriter : ISagaScriptWriter
+{
+    TextWriter writer;
+    SagaDefinition saga;
+    string tableName;
+
+    public OracleSagaScriptWriter(TextWriter textWriter, SagaDefinition saga)
+    {
+        writer = textWriter;
+        this.saga = saga;
+        if (saga.TableSuffix.Length > 27)
+        {
+            throw new Exception($"Saga '{saga.TableSuffix}' contains more than 27 characters, which is not supported by SQL persistence using Oracle. Either disable Oracle script generation using the SqlPersistenceSettings assembly attribute, shorten the name of the saga, or specify an alternate table name by overriding the SqlSaga's TableSuffix property.");
+        }
```

Comment:
Oracle requires <= 30. We are going to take 3 for the addition of `_PK` for the primary key name. I was trying to tell the user exactly what their limitation is rather than explain all the whys, which should be covered by doco. Open to better suggestions though.

> Either disable Oracle script generation using the SqlPersistenceSettings assembly attribute

It could be that you are using MSSQL/MySQL and so far you have short saga names, and you haven't disabled Oracle script generation. Now you add a 28 character saga name - you don't really need to shorten your saga name for MSSQL, you just need to stop generating the Oracle scripts you don't need.